tretools/phenotype_report/engine.py

```diff
@@ -189,3 +189,51 @@ def generate_empty_template_file(path: str) -> None:
             writer = csv.writer(csvfile)
             writer.writerow(columns)
 
+    def adding_DoB_Gender(self, GenderFilepath, GenderFile_OrageneID, GenderFile_PseudoNHS, GenderFile_Gender, DoBFile_Filepath, DoBFile_OrageneID, DoBFile_DoB):
+        pseudoNHSCol = 'PS_NHS'
+        codecol = 'Code'
+        unaccepted_PseudoNHS = ['UnfindableNHSnumber' , 'Studywithdrawal/IncompleteConsent/OtherIssue']
+
+        # Loading Gender file
+        GenderFilecols = (GenderFile_OrageneID, GenderFile_PseudoNHS, GenderFile_Gender)
+        # separator = self, separator(Genderfilepath) For this txt file, the separator is \t while for NHSD data it is ',' which is obtained by the dictionary of separators.
```

Comment:
To discuss next time. I have opened up a different pull request #7 to hopefully simplify this.